File: C/Lista9/Pojazd.scala
class Pojazd(val producer: String, val model: String, val productionYear: Int = -1, var registrationNumber: String = "")

4 konstruktory:
var p1 = new Pojazd("vw", "golf 1")
var p1 = new Pojazd("vw", "golf 1", 1990)
var p1 = new Pojazd("vw", "golf 1", 1990, "HFP")
var p1 = new Pojazd("vw", "golf 1", registrationNumber = "HFP") // argument nazwany (podawany jako ostatni)

File: C/Lista9/Pojazd.java
public class Pojazd {
	private String producer;
	private String model;
	private String registrationNumber;
	private int productionYear;
	
	public Pojazd(String producer, String model) {
		this(producer, model, "", -1);
	}
	
	public Pojazd(String producer, String model, String registrationNumber) {
		this(producer, model, registrationNumber, -1);
	}
	
	public Pojazd(String producer, String model, int productionYear) {
		this(producer, model, "", productionYear);
	}
	
	public Pojazd(String producer, String model, String registrationNumber, int productionYear) {
		this.producer = producer;
		this.model = model;
		this.registrationNumber = registrationNumber;
		this.productionYear = productionYear;
	}
	
	public String getProducer() {
		return producer;
	}
	
	public String getModel() {
		return model;
	}
	
	public int getProductionYear() {
		return productionYear;
	}
	
	public String getRegistrationNumber() {
		return registrationNumber;
	}
	
	public void setRegistrationNumber(String registrationNumber) {
		this.registrationNumber = registrationNumber;
	}
}